src/coreComponents/fileIO/Table/TableFormatter.cpp

```diff
@@ -0,0 +1,400 @@
+  {
+    integer newStringSize = extraCharactersPerColumn + columns[idxColumn].m_maxStringSize.size();
+    if( idxColumn == columns.size() - 1 )
+    {
+      columns[idxColumn].m_maxStringSize = GEOS_FMT( "{:>{}}",
+                                                     columns[idxColumn].m_maxStringSize,
+                                                     newStringSize + m_tableLayout.getColumnMargin() );
+    }
+    else
+    {
+      columns[idxColumn].m_maxStringSize = GEOS_FMT( "{:>{}}",
+                                                     columns[idxColumn].m_maxStringSize,
+                                                     newStringSize );
+    }
+  }
+}
+
+void TableTextFormatter::computeTableMaxLineLength( std::vector< TableLayout::Column > & columns,
+                                                    std::vector< string > const & msgTableError ) const
+{
+  integer const columnMargin = m_tableLayout.getColumnMargin();
+  integer const borderMargin = m_tableLayout.getBorderMargin();
+  string const tableTitle = string( m_tableLayout.getTitle() );
+
+  string::size_type sectionlineLength = ( ( columns.size() - 1 ) *  columnMargin ) + (borderMargin * 2);
+  string::size_type maxTopLineLength =  tableTitle.length();
+  string::size_type msgTableErrorLength = borderMargin;
+
+  if( !msgTableError.empty() )
+  {
+    auto maxStringSize = *(std::max_element( msgTableError.begin(), msgTableError.end(),
+                                             []( const auto & a, const auto & b ) {
+      return a.size() < b.size();
+    } ));
+
+    msgTableErrorLength += maxStringSize.size() + 1; // +1 for \n set later
+    maxTopLineLength = std::max( maxTopLineLength, msgTableErrorLength );
+  }
+
+  sectionlineLength += std::accumulate( columns.begin(), columns.end(), 0,
+                                        []( auto sum, const auto & column )
+  { return sum + column.m_maxStringSize.length();} );
+
+  if( sectionlineLength < maxTopLineLength )
+  {
+    integer const extraCharacters = maxTopLineLength - sectionlineLength;
+    increaseColumnsSize( columns, extraCharacters );
+  }
+}
+
+void TableTextFormatter::buildTableSeparators( std::vector< TableLayout::Column > const & columns,
+                                               string & topSeparator,
+                                               string & sectionSeparator ) const
+{
+  integer const columnMargin = m_tableLayout.getColumnMargin();
+  integer const borderMargin = m_tableLayout.getBorderMargin();
+
+  std::vector< string > maxStringPerColumn;
+
+  string const spaceBetweenColumns = GEOS_FMT( "{:-^{}}", verticalLine, columnMargin );
+
+  sectionSeparator = GEOS_FMT( "{}{:-<{}}", verticalLine, "", borderMargin );
+  for( auto const & col : columns )
+  {
+    maxStringPerColumn.push_back( string( integer( col.m_maxStringSize.length()), '-' ));
+  }
+
+  sectionSeparator += maxStringPerColumn[0];
+
+  for( size_t i = 1; i < maxStringPerColumn.size(); i++ )
```

Comment:
I think either you code could have more in-line comments or be more self-explaining, because it is quite complicated to follow what your are doing here.
Could "stringutilities::join()" be useful ?

Reply:
cant' be use in this situation